tests/e2e/tests/test_accommodation_v2.go

```diff
@@ -0,0 +1,449 @@
+// Copyright (C) 2022-2025, Chain4Travel AG. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package tests
+
+import (
+	"context"
+	"strconv"
+	"testing"
+	"time"
+
+	accommodationv2 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/services/accommodation/v2"
+	bookv2 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/services/book/v2"
+	typesv1 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/types/v1"
+	typesv2 "buf.build/gen/go/chain4travel/camino-messenger-protocol/protocolbuffers/go/cmp/types/v2"
+	"github.com/chain4travel/camino-messenger-bot/internal/metadata"
+	botGenerated "github.com/chain4travel/camino-messenger-bot/internal/rpc/generated"
+	"github.com/chain4travel/camino-messenger-bot/tests/e2e/bot"
+	partnerplugin "github.com/chain4travel/camino-messenger-bot/tests/e2e/partner_plugin"
+	"github.com/stretchr/testify/require"
+
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+/* Setting up the basic applications and services used in all sub-test-cases */
+func testAccommodationV2Setup(ctx context.Context, t *testing.T, tt *Test) (*partnerplugin.PartnerPlugin, *bot.Bot, *bot.Bot) {
+	// Register all the services needed for the tests
+	registerServices := []string{
+		botGenerated.AccommodationProductListServiceV2,
+		botGenerated.AccommodationProductInfoServiceV2,
+		botGenerated.AccommodationSearchServiceV2,
+		botGenerated.ValidationServiceV2,
+		botGenerated.MintServiceV2,
+	}
+	require.NoError(t, tt.caminoNetwork.Client.RegisterCMServices(ctx, registerServices))
+	supplierPartnerPlugin := tt.CreatePartnerPlugin(ctx, t)
+
+	supplierServices := []bot.CMService{
```

Comment:
I'd inline this into CreateBot so it would be closer to context

Reply:
Done